### PR TITLE
99dots fdc updates

### PR DESCRIPTION
**Pull request by proteusvacuum**
Updates to the 99DOTS Forwarders for speedy FDC rollout.

- Increases the amount of information we send them with every patient registration
- Sends adherence cases collected in enikshay if the person is registered in 99DOTS
- Sends a treatment outcome if one is collected.

https://docs.google.com/document/d/1PH4EPcSqhsYP-lV-_4_V6C9AvkpaBmIEe5ID5U2mAlc/edit#

Lots of lines, but much of it is boilerplate 😢 

@NoahCarnahan 
code buddies: @czue @kaapstorm

### CODE REVIEW

File: custom/enikshay/integrations/nikshay/tests/test_repeaters.py [hunk 1]
```diff
@@ -135,7 +133,7 @@ def test_payload_properties(self):
         self.assertEqual(payload['regBy'], "tbu-dmdmo01")
 
         # From Person
-        self.assertEqual(payload['pname'], "Pippin")
+        self.assertEqual(payload['pname'], "Peregrine Took")
```
Comment (czue) [inline custom/enikshay/integrations/nikshay/tests/test_repeaters.py]: important change.

File: custom/enikshay/integrations/ninetyninedots/repeater_generators.py [hunk 2]
```diff
@@ -132,6 +181,103 @@ def handle_failure(self, response, person_case, repeat_record):
             )
 
 
+@RegisterGenerator(NinetyNineDotsAdherenceRepeater, 'case_json', 'JSON', is_default=True)
+class AdherencePayloadGenerator(BasePayloadGenerator):
```
Comment (czue) [inline custom/enikshay/integrations/ninetyninedots/repeater_generators.py]: not sure if/where these are documented? could be nice for these each to have a doc string indicating what purpose they are serving.

File: custom/enikshay/integrations/ninetyninedots/tests/test_repeaters.py [hunk 3]
```diff
@@ -168,27 +182,105 @@ def test_create_person_no_episode(self):
...
+        self.repeater.save()
+
+    @run_with_all_backends
+    def test_trigger(self):
+        self.create_case_structure()
+        self._create_99dots_registered_case()
+        self._create_99dots_enabled_case()
+        self.assertEqual(0, len(self.repeat_records().all()))
+
+        self.create_adherence_cases([datetime(2017, 02, 17)])
```
Comment (czue) [inline custom/enikshay/integrations/ninetyninedots/tests/test_repeaters.py]: nit - these are octal `02`'s. really only an issue in august and september though.

**Review by czue — APPROVED** [review]
looks good! didn't review super closely, but assume that's fine and appreciate the test coverage.

looks like a real build failure
```
======================================================================
FAIL: test_payload_properties (custom.enikshay.integrations.nikshay.tests.test_repeaters.TestNikshayRegisterPatientPayloadGenerator)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 264, in inner
    return helper(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 237, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 252, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 233, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/custom/enikshay/integrations/nikshay/tests/test_repeaters.py", line 150, in test_payload_properties
    self.assertEqual(payload['dotname'], 'awesome dot')
AssertionError: u'Gandalf The Grey' != 'awesome dot'
-------------------- >> begin captured stdout << ---------------------
<function test_payload_properties at 0x7f02ca717ed8> failed with the following settings:
settings.TESTS_SHOULD_USE_SQL_BACKEND = False
```

File: custom/enikshay/integrations/ninetyninedots/tests/test_repeaters.py [hunk 4]
```diff
@@ -293,3 +384,93 @@ def test_handle_failure(self):
...
         )
+
+
+class TestAdherencePayloadGenerator(TestPayloadGeneratorBase):
+    def _get_actual_payload(self, casedb):
+        return AdherencePayloadGenerator(None).get_payload(None, casedb['adherence'])
+
+    @run_with_all_backends
+    def test_get_payload(self):
+        date = datetime(2017, 02, 20)
```
Comment (NoahCarnahan) [inline custom/enikshay/integrations/ninetyninedots/tests/test_repeaters.py]: octal. Evaluates to `2` in this case, though, so doesn't really matter.

File: custom/enikshay/integrations/ninetyninedots/tests/test_repeaters.py [hunk 5]
```diff
@@ -293,3 +384,93 @@ def test_handle_failure(self):
...
+                "beneficiary_id": "person",
+                "adherence_source": "99DOTS",
+                "adherence_date": "2017-02-20"
+            }
+        )
+        self.assertEqual(self._get_actual_payload(cases), expected_payload)
+
+    @run_with_all_backends
+    def test_handle_success(self):
+        date = datetime(2017, 02, 20)
```
Comment (NoahCarnahan) [inline custom/enikshay/integrations/ninetyninedots/tests/test_repeaters.py]: octal again

File: custom/enikshay/tests/test_case_utils.py [hunk 6]
```diff
@@ -124,3 +127,34 @@ def test_get_open_episode_case_from_occurrence(self):
             get_open_episode_case_from_occurrence(self.domain, self.occurrence_id).case_id,
             self.episode_id
         )
+
+    @run_with_all_backends
+    def test_get_episode_case_from_adherence(self):
+        adherence_case = self.create_adherence_cases([datetime(2017, 02, 17)])[0]
```
Comment (NoahCarnahan) [inline custom/enikshay/tests/test_case_utils.py]: octal

**Review by NoahCarnahan — APPROVED** [review]
lgtm, just one nit-pick

**Pull request merged**